lib/rubocop/cop/rspec/sort_metadata.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Sort RSpec metadata alphabetically.
+      #
+      # @example
+      #   # bad
+      #   describe 'Something', :b, :a
+      #   context 'Something', foo: 'bar', baz: true
+      #   it 'works', :b, :a, foo: 'bar', baz: true
+      #
+      #   # good
+      #   describe 'Something', :a, :b
+      #   context 'Something', baz: true, foo: 'bar'
+      #   it 'works', :a, :b, baz: true, foo: 'bar'
+      #
+      class SortMetadata < Base
+        extend AutoCorrector
+        include RangeHelp
+
+        HOOKS = %i[
+          prepend_before
+          before
+          append_before
+          around
+          prepend_after
+          after
+          append_after
+        ].to_set.freeze
+
+        MSG = 'Sort metadata alphabetically.'
+
+        # Examples, ExampleGroups, SharedGroups, and Hooks
+        RESTRICT_ON_SEND = %i[
+          after
+          append_after
+          append_before
+          around
+          before
+          context
+          describe
+          example
+          example_group
+          fcontext
+          fdescribe
+          feature
+          fexample
+          ffeature
+          fit
+          focus
+          fscenario
+          fspecify
+          it
+          its
+          pending
+          prepend_after
+          prepend_before
+          scenario
+          shared_context
+          shared_examples
+          shared_examples_for
+          skip
+          specify
+          xcontext
+          xdescribe
+          xexample
+          xfeature
+          xit
+          xscenario
+          xspecify
+        ].to_set.freeze
+
+        # @!method rspec_metadata(node)
+        def_node_matcher :rspec_metadata, <<~PATTERN
+          (send #rspec? %RESTRICT_ON_SEND _ $sym* (hash $...)?)
+        PATTERN
+
+        # @!method rspec_configure(node)
+        def_node_matcher :rspec_configure, <<~PATTERN
+          (block (send #rspec? :configure) (args (arg $_)) ...)
+        PATTERN
+
+        # @!method metadata_in_block(node)
+        def_node_search :metadata_in_block, <<~PATTERN
+          (send (lvar $_) %HOOKS _ $sym* (hash $...)?)
+        PATTERN
+
+        def on_block(node)
+          rspec_configure(node) do |block_var|
+            metadata_in_block(node).each do |receiver, symbols, pairs|
+              investigate(symbols, pairs.flatten) if receiver == block_var
+            end
+          end
+        end
+
+        def on_send(node)
+          rspec_metadata(node) do |symbols, pairs|
+            investigate(symbols, pairs.flatten)
+          end
+        end
+
+        private
+
+        def investigate(symbols, pairs)
+          return if symbols.none? && pairs.none?
```

Comment:
If `symbols.none? && pairs.none?` then line 108 will also return, right? So perhaps we can just remove line 107 altogether?